server/service/osquery.go

```diff
@@ -1973,6 +1975,19 @@ func (svc *Service) processVPPForNewlyFailingPolicies(
 			continue
 		}
 
+		scoped, err := svc.ds.IsVPPAppLabelScoped(ctx, vppMetadata.VPPAppTeam.ID, hostID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "checking if vpp app is label scoped to host")
+		}
+
+		if !scoped {
```

Comment:
With this in here, we need to make sure the VPP automation check calling this fn happens after the script automation check (I should've anticipated this in that PR, sorry) because while it's not possible to spec both a VPP and a normal installer automation on a policy, it _is_ possible to spec both installer and script automations on the same policy.

This means that, as things are right now (VPP evaluated before script), an out-of-scope VPP installer attached to a policy would cause a script automation to never run.

Now, moving things around I believe would cause the script to run every time the policy failed, but that's...probably a better bug to have than a no-op that only shows up on VPP and not custom packages/FMAs?

Reply:
@iansltx i just tested it out; the script does run weirdly enough, but the policy shows up as failing on the host details. I can try it out with the ordering changed

Reply:
Got it. Yeah, guessing flipping ordering will get the script run plus the desired nil policy status. If that doesn't fix it, we'll need to figure out why the failed status makes it back.

Reply:
@iansltx yep, changing the ordering worked (script runs, but policy status remains nil)

Reply:
writing up a test to track this case

Reply:
@iansltx test is in. will request re-review once CI is done.